modules/extension/xsd/xsd-core/src/main/java/org/geotools/xsd/impl/BindingPropertyExtractor.java

```diff
@@ -59,185 +72,192 @@ public void setContext(MutablePicoContainer context) {
+                        // be dangerous. What we shold do is force the schema to
+                        // resolve all of it simports when the encoder starts
+                        elementDecl =
+                                encoder.getSchema()
+                                        .resolveElementDeclaration(
+                                                name.getNamespaceURI(), name.getLocalPart());
                     }
 
-                    if (Schemas.isBaseType(elementDecl, el)) {
-                        reference = particle;
-                        break;
+                    // look for a particle in the containing type which is either
+                    // a) a base type of the element
+                    // b) in the same subsittuion group
+                    // if found use the particle to dervice multiplicity
+                    XSDParticle reference = null;
+                    for (Iterator p =
+                                    Schemas.getChildElementParticles(element.getType(), true)
+                                            .iterator();
+                            p.hasNext(); ) {
+                        XSDParticle particle = (XSDParticle) p.next();
+                        XSDElementDeclaration el = (XSDElementDeclaration) particle.getContent();
+                        if (el.isElementDeclarationReference()) {
+                            el = el.getResolvedElementDeclaration();
+                        }
+
+                        if (Schemas.isBaseType(elementDecl, el)) {
+                            reference = particle;
+                            break;
+                        }
                     }
-                }
 
-                // wrap the property in a particle
-                XSDParticle particle = XSDFactory.eINSTANCE.createXSDParticle();
-                XSDElementDeclaration wrapper = XSDFactory.eINSTANCE.createXSDElementDeclaration();
-                wrapper.setResolvedElementDeclaration(elementDecl);
-                particle.setContent(wrapper);
-                // particle.setContent(elementDecl);
-
-                // if there is a reference, derive multiplicity
-                if (reference != null) {
-                    particle.setMaxOccurs(reference.getMaxOccurs());
-                } else {
-                    // dervice from collection
-                    if (values.size() > 1) {
-                        // make a multi property
-                        particle.setMaxOccurs(-1);
+                    // wrap the property in a particle
+                    XSDParticle particle = XSDFactory.eINSTANCE.createXSDParticle();
+                    XSDElementDeclaration wrapper =
+                            XSDFactory.eINSTANCE.createXSDElementDeclaration();
+                    wrapper.setResolvedElementDeclaration(elementDecl);
+                    particle.setContent(wrapper);
+                    // particle.setContent(elementDecl);
```

Comment:
Letfover?

Reply:
Looks like original code leftover, of course we can clean it.